File: chapter12end/java/aircraft-positions/src/main/java/com/thehecklers/aircraftpositions/AircraftRepository.java
package com.thehecklers.aircraftpositions;

import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;

public interface AircraftRepository extends ReactiveCrudRepository<Aircraft, Long> {
    Flux<Aircraft> findAircraftByReg(String reg);
}

File: chapter12end/kotlin/aircraft-positions/src/main/kotlin/com/thehecklers/aircraftpositions/AircraftRepository.kt
package com.thehecklers.aircraftpositions

import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux

interface AircraftRepository: ReactiveCrudRepository<Aircraft, String> {
    fun findAircraftByReg(reg: String): Flux<Aircraft>
}